client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-post-setup/index.tsx

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+import { StepContainer, base64ImageToBlob, uploadAndSetSiteLogo } from '@automattic/onboarding';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+import { FormEvent, useEffect, useState } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SITE_STORE } from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSite } from '../../../../hooks/use-site';
+import SetupForm from '../link-in-bio-setup/setup-form';
+import type { Step } from '../../types';
+
+import '../link-in-bio-setup/styles.scss';
+
+const LinkInBioPostSetup: Step = function LinkInBioPostSetup( { navigation } ) {
+	const { submit } = navigation;
+	const { __ } = useI18n();
+	const site = useSite();
+
+	const [ siteTitle, setComponentSiteTitle ] = useState( '' );
+	const [ tagline, setTagline ] = useState( '' );
+	const [ invalidSiteTitle, setInvalidSiteTitle ] = useState( false );
+	const [ selectedFile, setSelectedFile ] = useState< File | undefined >();
+	const [ base64Image, setBase64Image ] = useState< string | null >();
+	const [ isLoading, setIsLoading ] = useState< boolean >( false );
+
+	const { saveSiteSettings } = useDispatch( SITE_STORE );
+
+	useEffect( () => {
+		setComponentSiteTitle( site?.name || '' );
+		setTagline( site?.description || '' );
+	}, [ site ] );
+
+	useEffect( () => {
+		setIsLoading( false );
```

Comment:
This will enable the button if you type something during loading. I think we should reset the loading state at more specific places (when loading is done).

To see this, add 5000ms delay in DevTools network and submit, then change the title.

Reply:
Yes, this was what I originally had been doing. I've adjusted this back so that the loading state is updated when as soon as loading is done. This does mean the 'Continue' button becomes active for just a moment prior to page redirect, which can feel a bit jumpy from a UI perspective. But accurate handling of the loading stat is probably more important here.

Reply:
It won't be jumpy because React will batch the two calls since they're inside an event handler.